### ISSUE TITLE
Mobile Support (IOS/Android)

### ISSUE THREAD

**Issue by emilio-toledo** [mnaoumov/obsidian-fix-require-modules#1]
Hello, thank you so much for this plugin!

I have one question, is it possible to get mobile support?

I have found "fix-require-modules" extremely helpful, and I believe adding mobile support would greatly its popularity.

**Comment by mnaoumov** [comment]
@emilio-toledo what is scenario you are looking for using the plugin on Mobile?

**Comment by emilio-toledo** [comment]
I am currently using [dataviewjs](https://github.com/blacksmithgu/obsidian-dataview) to render the following snippet:
```js
const { init } = require("/views/pitch.ts")
const data = init(dv)
```
This snippet invokes a file which renders custom HTML to display Japanese Kana in a specific way and style. Not only that but all throughout my vault I am using this module to import .ts scripts to add a bunch of extra functionality when paired with [templater](https://github.com/SilentVoid13/Templater), such as modifying the frontmatter data on template load and fetching data using `fetch`.

My whole vault and ideas that I have to enhance my Obsidian workflow without having to develop a plugin from scratch depend on custom JS/TS. I have tried most of the plugins related to creating modules and this one is by far the best.

**Comment by mnaoumov** [comment]
@emilio-toledo That's very unlikely we could make the plugin work on Mobile because on Mobile we don't have `Node.js` infrastructure. All the plugins that aim to run scripts on Mobile don't use `require()`, they use `eval()`. But I will double check

**Comment by emilio-toledo** [comment]
Maybe this is out of the scope for this plugin, but what [CustomJS](https://github.com/saml-dev/obsidian-custom-js) is doing is adding a global function that returns your modules (they export classes). Maybe it would be possible that depending on the platform, we either use require or circumvent mobile limitations by using eval in a similar way.

**Comment by mnaoumov** [comment]
@emilio-toledo I know for a fact, `CustomJS` is using `eval()` inside. I was a contributor to `CustomJS` for a while and decided to write my own version of it to do things that `eval()` can't

**Comment by mnaoumov** [comment]
I added a very basic support for mobile for things like

 ```js
require('obsidian');
require('@codemirror/view');
require('obsidian/app');
```

Unfortunately adding anything more serious is unlikely to be possible due to the [limitations](https://github.com/mnaoumov/obsidian-fix-require-modules?tab=readme-ov-file#mobile-support)

I am closing this issue for now

**Comment by mnaoumov** [comment]
@emilio-toledo thanks for your input. After thinking for a while I found a way to add this functionality to mobile.

In https://github.com/mnaoumov/obsidian-fix-require-modules/releases/tag/8.0.0 everything for mobile is also supported with some trick mentioned in the documentation
